source/common/http/conn_manager_impl.cc

```diff
@@ -1352,23 +1352,40 @@ void ConnectionManagerImpl::ActiveStream::sendLocalReply(
     createFilterChain();
   }
   stream_info_.setResponseCodeDetails(details);
+
   Utility::sendLocalReply(
-      is_grpc_request,
-      [this, modify_headers](HeaderMapPtr&& headers, bool end_stream) -> void {
-        if (modify_headers != nullptr) {
-          modify_headers(*headers);
-        }
-        response_headers_ = std::move(headers);
-        // TODO: Start encoding from the last decoder filter that saw the
-        // request instead.
-        encodeHeaders(nullptr, *response_headers_, end_stream);
-      },
-      [this](Buffer::Instance& data, bool end_stream) -> void {
-        // TODO: Start encoding from the last decoder filter that saw the
-        // request instead.
-        encodeData(nullptr, data, end_stream, FilterIterationStartState::CanStartFromCurrent);
-      },
-      state_.destroyed_, code, body, grpc_status, is_head_request);
+      state_.destroyed_,
+      Utility::EncodeFunctions{
+          [this, modify_headers](HeaderMapPtr&& headers, bool end_stream) -> void {
+            if (modify_headers != nullptr) {
+              modify_headers(*headers);
+            }
+            response_headers_ = std::move(headers);
+            // TODO: Start encoding from the last decoder filter that saw the
+            // request instead.
+            encodeHeaders(nullptr, *response_headers_, end_stream);
+          },
+          [this](Buffer::Instance& data, bool end_stream) -> void {
+            // TODO: Start encoding from the last decoder filter that saw the
+            // request instead.
+            encodeData(nullptr, data, end_stream, FilterIterationStartState::CanStartFromCurrent);
+          }},
+      Utility::RewriteReplyFunctions{
+          [this](Code& code) -> void {
+            connection_manager_.config_.localReply()->matchAndRewrite(
+                request_headers_.get(), response_headers_.get(), response_trailers_.get(),
```

Comment:
Currently it is possible that `*_headers_` and `stream_info_` are null. I want to run local reply mapper after headers are available for filters/formatter. Should I validate if values are not null, or there should be change that makes headers and stream_info initialized before local reply mapper functions are called?  Looking for some suggestions/advices how to handle this case.